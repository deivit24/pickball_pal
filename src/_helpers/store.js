import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
