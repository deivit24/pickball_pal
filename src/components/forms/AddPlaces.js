// React
import React, { useState } from 'react';
// Redux
import { useDispatch } from 'react-redux';
// React Router Dom
import { useHistory } from 'react-router-dom';
// Add Place action
import { addPlaces } from '../../_actions/profile';
// Material UI Components
import { TextField, Button } from '@material-ui/core';

import '../../assets/css/AddPlaces.css';

const INIT_STATE = {
  name: '',
  location: '',
};
const AddPlaces = () => {
  const history = useHistory();
  const dispatch = useDispatch();
  const [values, setValues] = useState(INIT_STATE);

  const { name, location } = values;

  const handleChange = (e) => {
    const { name, value } = e.target;
    setValues((l) => ({ ...l, [name]: value }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(addPlaces(values, history));
  };
  return (
    <div id="AddPlaces" className="container">
      <div className="form-box">
        <div className="form-title">
          <h2>Add Place</h2>
        </div>
        <form noValidate autoComplete="on" onSubmit={handleSubmit}>
          <div className="row">
            <div className="col-12">
              <TextField
                id="outlined-full-width"
                label="Name"
                name="name"
                value={name}
                onChange={handleChange}
                helperText="Name of your location (eg. Westlake HS)"
                fullWidth
                margin="normal"
                variant="outlined"
              />
            </div>

            <div className="col-12">
              <TextField
                id="outlined-full-width"
                label="Location"
                name="location"
                value={location}
                onChange={handleChange}
                helperText="Location (eg. Westlake High School, Austin, TX)"
                fullWidth
                margin="normal"
                variant="outlined"
              />
            </div>

            <Button
              onClick={handleSubmit}
              variant="contained"
              color="primary"
              className="submit-button"
            >
              Submit
            </Button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default AddPlaces;
