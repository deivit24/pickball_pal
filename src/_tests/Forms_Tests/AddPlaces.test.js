import React from 'react';
import { render } from '@testing-library/react';
import AddPlaces from '../../components/forms/AddPlaces';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing AddPlaces

describe('Testing the AddPlaces', () => {
  it('it renders AddPlaces without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <AddPlaces />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <AddPlaces />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
