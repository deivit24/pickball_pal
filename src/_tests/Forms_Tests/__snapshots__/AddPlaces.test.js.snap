// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the AddPlaces matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    id="AddPlaces"
  >
    <div
      class="form-box"
    >
      <div
        class="form-title"
      >
        <h2>
          Add Place
        </h2>
      </div>
      <form
        autocomplete="on"
        novalidate=""
      >
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-full-width"
                id="outlined-full-width-label"
              >
                Name
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-describedby="outlined-full-width-helper-text"
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-full-width"
                  name="name"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Name
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-contained"
                id="outlined-full-width-helper-text"
              >
                Name of your location (eg. Westlake HS)
              </p>
            </div>
          </div>
          <div
            class="col-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-full-width"
                id="outlined-full-width-label"
              >
                Location
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-describedby="outlined-full-width-helper-text"
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-full-width"
                  name="location"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Location
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-contained"
                id="outlined-full-width-helper-text"
              >
                Location (eg. Westlake High School, Austin, TX)
              </p>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained submit-button MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
